src/main/kotlin/net/irisshaders/lilybot/extensions/util/RoleMenu.kt

```diff
@@ -4,206 +4,523 @@ import com.kotlindiscord.kord.extensions.DISCORD_BLACK
-						description = arguments.description + descriptionAppendix
-						color = arguments.color
+					val message = channel.getMessageOrNull(arguments.messageId)
+					if (!roleMenuExists(message, arguments.messageId)) {
+						return@action
 					}
 
-					val components = components {
-						ephemeralButton(row = 0) {
-							label = "Add role"
-							style = ButtonStyle.Success
-							id = arguments.role.name + "add"
+					val data = DatabaseHelper.getRoleData(arguments.messageId)!!
 
-							val newComponent = ComponentData(
-								arguments.role.name + "add",
-								arguments.role.id,
-								"add"
-							)
-							DatabaseHelper.setComponent(newComponent)
+					if (arguments.role.id in data.roles) {
+						respond {
+							content = "This menu already contains that role."
+						}
+						return@action
+					}
 
-							action { }
+					if (data.roles.size == 24) {
+						respond {
+							content = "You can't have more than 24 roles in a role menu. This is a Discord limitation."
 						}
+						return@action
+					}
 
-						ephemeralButton(row = 0) {
-							label = "Remove role"
-							style = ButtonStyle.Danger
-							id = arguments.role.name + "remove"
+					data.roles.add(arguments.role.id)
+					DatabaseHelper.setRoleMenu(
+						data.messageId,
+						data.channelId,
+						data.guildId,
+						data.roles
+					)
 
-							val newComponent = ComponentData(
-								arguments.role.name + "remove",
-								arguments.role.id,
-								"remove"
-							)
-							DatabaseHelper.setComponent(newComponent)
+					val config = DatabaseHelper.getConfig(guild!!.id)
+					val actionLog = guild!!.getChannel(config!!.modActionLog) as GuildMessageChannelBehavior
 
-							action { }
+					actionLog.createMessage {
+						embed {
+							title = "Role Added to Role Menu"
+							description = "The ${arguments.role.mention} role was added to a role menu in " +
+									"${channel.mention}."
+						}
+						components {
+							linkButton {
+								label = "Jump to role menu"
+								url = message!!.getJumpUrl()
+							}
 						}
 					}
 
-					components.removeAll()
+					respond {
+						content = "Added the ${arguments.role.mention} role to the specified role menu."
+					}
+				}
+			}
 
-					respond { content = "Role menu created." }
+			ephemeralSubCommand(::RoleMenuRemoveArgs) {
+				name = "remove"
+				description = "Remove a role from the existing role menu in this channel."
 
-					// Try to get the action log from the config.
-					// If a config is not set, inform the user and return@action
-					val config = DatabaseHelper.getConfig(guild!!.id)!!
-					val actionLog = guild?.getChannel(config.modActionLog) as GuildMessageChannelBehavior
+				check {
+					anyGuild()
+					hasPermission(Permission.ManageMessages)
+					configPresent()
+				}
+
+				action {
+					val menuMessage = channel.getMessageOrNull(arguments.messageId)
+					if (!roleMenuExists(menuMessage, arguments.messageId)) {
+						return@action
+					}
```

Comment:
```suggestion
					if (!roleMenuExists(menuMessage, arguments.messageId)) return@action
```
Brackets aren't really needed, nor used elsewhere in the codebase in situations like this